velox/docs/functions/spark/string.rst

```diff
@@ -427,3 +427,25 @@ String Functions
     Returns string with all characters changed to uppercase. ::
 
         SELECT upper('SparkSql'); -- SPARKSQL
+
+.. spark:function:: varchar_type_write_side_check(string, limit) -> varchar
+
+    Removes trailing space characters (ASCII 32) that exceed the maximum Unicode string length ``limit`` in characters.
+    This function will trim at most the length of ``string`` in characters - ``limit`` space characters from the end of this string.
+    Throws exception when the string still exceeds ``limit`` after trimming trailing spaces or when ``limit`` is negative.
+    Note: This function is not directly callable in SQL, but internally used by SparkSQL for length check when writing to
+    varchar type columns. ``limit=0`` is technically valid for this function but SparkSQL does not support ``VARCHAR(0)``
```

Comment:
> ``limit=0`` is technically valid for this function but SparkSQL does not support ``VARCHAR(0)`` in DDL. This case exists primarily for internal consistency.

This description is a bit confusing from my view. Perhaps just clarify what's is output of limit being 0, and whether it is aligned with Spark.